File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        
        int n = nums.size();
        
        vector<int>ans(n);
        
        int i = 0, j = nums.size() - 1, k = j;
        
        while(i <= j){
            
            if(abs(nums[i]) > abs(nums[j])){
                ans[k--] = (nums[i] * nums[i]);
                ++i;
            }
            else {   
                ans[k--] = (nums[j] * nums[j]);
                --j;
            }
            
        }
        
        return ans;        
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        
        for(int i = 0; i < n; i++){
            nums[i] = (nums[i] * nums[i]);
        }
        
        int res[] = new int[n];
        int i = 0, j = n - 1, k = n - 1;
        
        while(i <= j){
            if(nums[i] >= nums[j]){
                res[k--] = nums[i++];
            }else{
                res[k--] = nums[j--];
            }
        }
        
        return res;
    }
}